webknossos/webknossos/geometry/mag.py

```diff
@@ -1,87 +1,112 @@
 import re
 from functools import total_ordering
 from math import log2
-from typing import Any, List
+from typing import Any, List, Optional, Tuple, cast
 
+import attr
 import numpy as np
 
+from .vec3_int import Vec3Int, Vec3IntLike
+
+
+def import_mag(mag_like: Any) -> Vec3Int:
+    as_vec3_int: Optional[Vec3Int] = None
+
+    if isinstance(mag_like, int):
+        as_vec3_int = Vec3Int(mag_like, mag_like, mag_like)
+    elif isinstance(mag_like, Vec3Int):
+        as_vec3_int = mag_like
+    elif isinstance(mag_like, list) or isinstance(mag_like, tuple):
+        as_vec3_int = Vec3Int(cast(Vec3IntLike, mag_like))
+    elif isinstance(mag_like, str):
+        if re.match(r"^\d+$", mag_like) is not None:
+            as_vec3_int = Vec3Int(int(mag_like), int(mag_like), int(mag_like))
+        elif re.match(r"^\d+-\d+-\d+$", mag_like) is not None:
+            as_vec3_int = Vec3Int([int(m) for m in mag_like.split("-")])
+    elif isinstance(mag_like, Mag):
+        as_vec3_int = Vec3Int(mag_like.to_tuple())
```

Comment:
We could directly return `mag_like` at the start of the function in this case to ensure `Mag(Mag(5))` is as simple as possible, similar as with Vec3Int.

Reply:
The import method needs to return a Vec3IntLike, not a Mag, since this is not the constructor but the converter for the attr.ib. Still, I moved up the `Mag` case in the switch, and removed the superfluous tuple conversion